### PR TITLE
Antag Rebalances, Armory Restock

### PR DESCRIPTION
**Pull request by TheDanseMacabre**
Essentially, the aim of this PR is to:
1. Make squishy antags like traitors more resilient. The biggest thing that makes security so much stronger than antags is that security has access to medbay and the comprehensive medical care medbay can give them.
2. Make security's armory better by expanding their options in regards to round-start weapons available.

To that end, I have added Sanasomnum, a drug capable of healing almost any injury and healing it quite quickly, though it makes the user unconscious for a few minutes. Things that would normally require surgery or otherwise intensive medical care can all be fixed. Some antags also have their TC counts increased to 20, which would allow them to buy two sanasomnum injectors while leaving the previous level of TCs for normal use, plus one. Cultist robes are also better armored against ballistics, at the cost of some laser armor. Raiders now get an uplink to allow them some gimmick versatility and access to sanasomnum, but with only 3 TCs each.

Security, on the other hand, will receive machine pistols, buckshot, an a ballistic carbine loaded with low-damage polymer bullets. The machine pistols and the buckshot are all easily stopped by pistol-level ballistic armor. The carbine, on the other hand, can penetrate up to rifle level armor - but it does not go through targets and only does 25 damage, which I have seen mitigated to around 15 by uplink heavy armor.

This PR also buffs and slightly reworks horror form. Previously it was totally broken (only had 250 health because of a bug nobody noticed, couldn't do enough damage to hurt anyone, ranged attack was nonfunctional.) The price has been lowered marginally, health increased, passive regen removed, upper damage limit increased considerably, etc. All in all, horror form is a real, credible threat now. Changelings will be able to enjoy a more adequate final boss.

There's a number of other minor antagonist buffs and additions. Some items have been added to the uplink, like the SAW, ballistic carbine, and other guns. Gear crates have been made available to all antags, and burglars no longer have arbitrary category restrictions. Random gun now costs 6 TCs, making it more of a challenge option.

Exact details can be found in the changelog.

My hope is that this will even the odds. Antags and security will be closer to being on the same footing, as security will have better weapons - and antags will have desperately needed healing items.

Extremely WIP. I don't really know what I'm doing, and this code may be embarrassing.

### CODE REVIEW

**Comment by TheDanseMacabre** [comment]
!wip

**Comment by Arrow768** [comment]
DNM due to the balance aspects which will be reviewed/discussed by the maintainers

**Comment by NonQueueingMatt** [comment]
I do not believe you are wrong, however most of the common antags (traitors) have exactly zero ways to deal with ballistic damage. You need to give them some way to deal with it.

**Comment by TheDanseMacabre** [comment]
!review

File: code/datums/uplink/ammunition.dm [hunk 1]
```diff
@@ -52,6 +52,12 @@
 	path = /obj/item/ammo_magazine/boltaction
 	desc = "Contains five rounds of 7.62mm bullets."
 
+/datum/uplink_item/item/ammo/carbine_mag
+	name = "5.56 carbine magazine"
+	item_cost = 2
+	path = /obj/item/ammo_magazine/a556/carbine 
+	desc = "Contains 15 rounds of 5.56"
```
Comment (NonQueueingMatt) [inline code/datums/uplink/ammunition.dm:59]:
```suggestion
	desc = "Contains 15 rounds of 5.56."
```

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Medicine.dm [hunk 2]
```diff
@@ -1306,6 +1306,69 @@
...
+		for(var/obj/item/organ/external/E in H.organs)
+			if(E.status & ORGAN_BROKEN && prob(10))
+				E.status &= ~ORGAN_BROKEN
+	if(M.chem_doses[type] > 5)
+		for(var/obj/item/organ/external/E in H.organs)
+			if(E.status & TENDON_CUT && prob(10))
+				E.status &= ~TENDON_CUT
+
+
+
```
Comment (NonQueueingMatt) [inline code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Medicine.dm:1371]: Remove this extra space.

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Medicine.dm [hunk 3]
```diff
@@ -1306,6 +1306,69 @@
...
+		return
+	var/dose = M.chem_doses[type]
+	if(dose == metabolism)
+		M.confused += 2
+		M.drowsiness += 2
+	else if(dose < 2)
+		M.Weaken(30)
+		M.eye_blurry = max(M.eye_blurry, 10)
+	else
+		M.sleeping = max(M.sleeping, 30)
```
Comment (NonQueueingMatt) [inline code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Medicine.dm:1356]: Duplicate effect - you already have CE_SEDATE given as a chem effect. Pick one of the two.

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Medicine.dm [hunk 4]
```diff
@@ -1306,6 +1306,69 @@
...
+		M.add_chemical_effect(CE_SEDATE, 30)
+		M.add_chemical_effect(CE_STABLE, 15)
+		M.add_chemical_effect(CE_UNDEXTROUS, 30)
+		M.add_chemical_effect(CE_CLUMSY, 30)
+
+/decl/reagent/sanasomnum/affect_blood(var/mob/living/carbon/M, var/alien, var/removed, var/datum/reagents/holder)
+	M.adjustCloneLoss(-20 * removed)
+	M.adjustOxyLoss(-2 * removed)
+	M.heal_organ_damage(20 * removed, 20 * removed)
+	M.adjustToxLoss(-1 * removed)
```
Comment (NonQueueingMatt) [inline code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Medicine.dm:1337]: Duplicate effect - keep either CE_ORGANREPAIR above or this one.

File: code/datums/outfits/outfit_antag.dm [hunk 5]
```diff
@@ -151,6 +151,7 @@
 	gloves = /obj/item/clothing/gloves/swat
 	shoes = /obj/item/clothing/shoes/jackboots
 	pda = /obj/item/modular_computer/handheld/pda/syndicate
+	r_pocket = /obj/item/device/special_uplink/mercenary
```
Comment (NonQueueingMatt) [inline code/datums/outfits/outfit_antag.dm:154]: Undocumented. Put it in the CL and PR description.

File: code/datums/outfits/outfit_antag.dm [hunk 6]
```diff
@@ -317,7 +318,7 @@
 	belt = null
 	gloves = null
 	l_ear = /obj/item/device/radio/headset/raider
-	l_pocket = /obj/item/device/contract_uplink
+	l_pocket = /obj/item/device/special_uplink/raider
```
Comment (NonQueueingMatt) [inline code/datums/outfits/outfit_antag.dm:321]: Undocumented. Put it in the CL and PR description.

File: code/game/objects/items/devices/uplink.dm [hunk 7]
```diff
@@ -411,9 +411,17 @@ A list of items and costs is stored under the datum of every game mode, alongsid
...
 	starting_telecrystals = 15
 
+/obj/item/device/special_uplink/mercenary
+	name = "milspec uplink"
+	starting_telecrystals = 15
+
 /obj/item/device/special_uplink/burglar
 	name = "sponsored uplink"
-	starting_telecrystals = 15
+	starting_telecrystals = 20
```
Comment (NonQueueingMatt) [inline code/game/objects/items/devices/uplink.dm:420]: Undocumented. Put it in the CL and PR description.

File: code/modules/reagents/reagent_containers/hypospray.dm [hunk 8]
```diff
@@ -306,3 +306,11 @@
 	name = "combat hypospray"
 	desc = "A sleek black hypospray. Its needle has the ability to bypass armor."
 	reagents_to_add = FALSE
+
+/obj/item/reagent_containers/hypospray/autoinjector/sanasomnum
+	name = "sanasomnum autoinjector"
+	desc = "A special autoinjector loaded with outlawed biomechanical stem cells, inducing a regenerative coma so intense it can heal almost any injury - even broken bones, organ and brain damage, severed tendons, and arterial damage. Upon use one will fall immediately into a state of unconsciousness lasting roughly three to five minutes, arising completely healed. The only thing it cannot fix are organs that have been destroyed outright, or so much cumulative damage that death is all but certain."
+	volume = 20
+	amount_per_transfer_from_this = 20
```
Comment (NonQueueingMatt) [inline code/modules/reagents/reagent_containers/hypospray.dm:314]: Do we really need 20 units at once of this chem?

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]
Instead of sanasomnum's current behaviour, you should consider making it sleep you for the current duration, instantly fix all ailments and then delete itself from the bloodstream to avoid any funny incidents with leftover chemical effects or drugs in the system. I do remember that you said you tested this extensively though, so it's not a hard requirement -- just something to think about.

File: html/changelogs/dansemacabre-gunsarefun.yml [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+# Optional: Remove this file after generating master changelog.  Useful for PR changelogs that won't get used again.
+delete-after: True
+
+# Any changes you've made.  See valid prefix list above.
+# INDENT WITH TWO SPACES.  NOT TABS.  SPACES.
+# SCREW THIS UP AND IT WON'T WORK.
+# Also, all entries are changed into a single [] after a master changelog generation. Just remove the brackets when you add new entries.
+# Please surround your changes in  double quotes ("), as certain characters otherwise screws up compiling. The quotes will not show up in the changelog.
+changes:
+  - rscadd: "Added a new reagent called Sanasomnum to the uplink for 2 TCs, which induces a regenerative coma that can heal almost any type of damage, including bone breaks, brain damage, ."
```
Comment (NonQueueingMatt) [inline html/changelogs/dansemacabre-gunsarefun.yml:41]:
```suggestion
  - rscadd: "Added a new reagent called Sanasomnum to the uplink for 2 TCs, which induces a regenerative coma that can heal almost any type of damage, including bone breaks, brain damage."
```

Comment (NonQueueingMatt) [reply html/changelogs/dansemacabre-gunsarefun.yml:41]: I think you didn't finish this sentence considering the wrong comma at the end.

File: code/modules/reagents/reagent_containers/hypospray.dm (hunk 8 above)
Comment (TheDanseMacabre) [reply code/modules/reagents/reagent_containers/hypospray.dm:314]: 10 is consumed far too quickly and furthermore I think giving people multiple hypos is a touch too strong.

**Comment by TheDanseMacabre** [comment]
> Instead of sanasomnum's current behaviour, you should consider making it sleep you for the current duration, instantly fix all ailments and then delete itself from the bloodstream to avoid any funny incidents with leftover chemical effects or drugs in the system. I do remember that you said you tested this extensively though, so it's not a hard requirement -- just something to think about.

I'm not a fan of this because it feels really game-y to me. Sanasomnum might be very similar to an adminheal, but it achieves this through healing instead of magical rejuvenation like the admin healing chem. I have tested this like you mentioned and the healing factor isn't strong enough to counteract injuries being actively sustained while healing.

**Review by JohnWildkins — APPROVED** [review]

**Review by NonQueueingMatt — APPROVED** [review]

File: code/modules/mob/living/simple_animal/hostile/changeling.dm [hunk 10]
```diff
@@ -1,6 +1,6 @@
 /mob/living/simple_animal/hostile/true_changeling
 	name = "shambling horror"
-	desc = "A monstrous creature, made of twisted flesh and bone."
+	desc = "An entity ripped from your nightmares. A monstrous creature, a warped parody of a living being. It is created from twisted flesh and bone, covered in oozing sores with bony protrusions sticking from them and serrated knife-like blades of bone. Looking at it makes you feel ill."
```
Comment (Arrow768) [inline code/modules/mob/living/simple_animal/hostile/changeling.dm:3]: We generally avoid "makes you ..." statements in descriptions

**Pull request merged**